test/mysql.autoupdate.test.js

```diff
@@ -788,6 +788,93 @@ describe('MySQL connector', function() {
     });
   });
 
+  it('should auto migrate/update custom foreign keys name with its constraint', function(done) {
+    var sourceModel = {
+      'name': 'SourceModel',
+      'options': {
+        'idInjection': false,
+        'mysql': {
+          'schema': 'myapp_test',
+          'table': 'source_model',
+        },
+      },
+      'properties': {
+        'id': {
+          'type': 'String',
+          'length': 20,
+          'id': 1,
+        },
+        'name': {
+          'type': 'String',
+          'length': 20,
+        },
+      },
+    };
+
+    var customFkModel = {
+      'name': 'CustomFkTest',
+      'options': {
+        'idInjection': false,
+        'mysql': {
+          'schema': 'myapp_test',
+          'table': 'custom_fk_test',
+        },
+        'foreignKeys': {
+          'fk_customfktest_customerId': {
+            'name': 'fk_customfktest_customerId',
+            'entity': 'SourceModel',
+            'entityKey': 'id',
+            'foreignKey': 'customerId',
+          },
+        },
+      },
+      'properties': {
+        'id': {
+          'type': 'String',
+          'length': 20,
+          'id': 1,
+        },
+        'customerId': {
+          'type': 'String',
+          'length': 20,
+          'mysql': {'columnName': 'custom_fk'},
+        },
+      },
+    };
+
+    var foreignKeySelect =
+    'SELECT COLUMN_NAME,CONSTRAINT_NAME,REFERENCED_TABLE_NAME, REFERENCED_COLUMN_NAME ' +
+    'FROM   INFORMATION_SCHEMA.KEY_COLUMN_USAGE ' +
+    'WHERE  REFERENCED_TABLE_SCHEMA = "myapp_test" ' +
+    'AND   TABLE_NAME = "custom_fk_test"';
+    var getCreateTable = 'SHOW CREATE TABLE `myapp_test`.`custom_fk_test`';
+
+    ds.createModel(sourceModel.name, sourceModel.properties, sourceModel.options);
+    ds.createModel(customFkModel.name, customFkModel.properties, customFkModel.options);
+
+    // do initial update/creation of table
+    ds.autoupdate(function(err) {
+      assert(!err, err);
+      ds.discoverModelProperties('custom_fk_test', function(err, props) {
+        // validate that we have the correct number of properties
+        assert.equal(props.length, 2);
```

Comment:
It's better to use `should` than `assert` as it produces more helpful error messages on failure.

```suggestion
        props.should.have.length(2);
```

The same comment applies to other assertions below too.